### PR TITLE
Refactor 'Hide shipping costs' Cart attribute logic

### LINKED ISSUE #1908: Cart 'Hide shipping costs' setting is ignored in the frontend
**To Reproduce**
1. Create a post with a _Cart_ block and set _Hide shipping costs_ option to true.
imatge
2. View the block in the frontend and notice it's ignored.
imatge

### PR DESCRIPTION
**Pull request by Aljullu**
False alarm: ~Fixes #1908.~

This PR was initially intended to fix the issue above :point_up_2: but actually that was not an issue, so I updated this PR to be just a refactor. Everything should work the same between `master` and this branch.

### How to test the changes in this Pull Request:

1. Create a post and add a _Cart_ block.
2. Try toggling the attributes and verify they persist in the editor between reloads and in the frontend.

### CODE REVIEW

File: assets/js/blocks/cart-checkout/cart/index.js [hunk 1]
```diff
@@ -66,8 +59,8 @@ const settings = {
 			isShippingCostHidden,
 		} = attributes;
 		const data = {
-			'data-isshippingcalculatorenabled': isShippingCalculatorEnabled,
-			'data-isshippingcosthidden': isShippingCostHidden,
+			'data-is-shipping-calculator-enabled': isShippingCalculatorEnabled,
+			'data-is-shipping-cost-hidden': isShippingCostHidden,
```
Comment (haszari) [inline assets/js/blocks/cart-checkout/cart/index.js]: Nice to see these attributes fixed up to use conventional camelCase 👍

File: assets/js/blocks/cart-checkout/cart/full-cart/index.js [hunk 2]
```diff
@@ -106,31 +105,18 @@ const Cart = ( {
...
-				return setShowShippingCosts( true );
-			}
-		}
-		return setShowShippingCosts( false );
-	}, [ isShippingCalculatorEnabled, isShippingCostHidden, shippingAddress ] );
+	const showShippingCosts = Boolean(
+		SHIPPING_ENABLED &&
+			isShippingCalculatorEnabled &&
+			( ! isShippingCostHidden || shippingAddress?.country )
+	);
```
Comment (haszari) [inline assets/js/blocks/cart-checkout/cart/full-cart/index.js]: Woah! This is much cleaner. Why did we need state and an effect beforehand?

**Comment by haszari** [comment]
@Aljullu Forgot to mention that this is currently breaking the e2e tests - because the block attributes changed.

If you've got e2e set up, you can regenerate the snapshot like so: `npm run test:e2e:update`, then check & commit the changed snapshot. This should be fixed before merge :)

**Comment by Aljullu** [comment]
Thanks for testing @haszari!

> I tested and saw an issue, which I'm guessing was already present in master (and maybe an quirk in core?). The `Hide shipping costs until an address is entered` option doesn't seem to do anything. 

Right, that's a bit confusing. In #1787 that will be improved. But for now, in order to get no default address you need to go to _WooCommerce_ > _Settings_ > _Default customer location_ and set it to _No location by default_.

> @Aljullu Forgot to mention that this is currently breaking the e2e tests - because the block attributes changed.

Missed that. :man_facepalming: Thank you!

**Pull request merged**